hbase-common/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileContext.java

```diff
@@ -138,6 +138,10 @@ public boolean isCompressedOrEncrypted() {
     return compressAlgo;
   }
 
+  public void setCompression(Compression.Algorithm compressAlgo) {
+    this.compressAlgo = compressAlgo;
+  }
+
```

Comment:
Whats going on here?

HFileContext is supposed to be ' Read-only HFile Context Information' but here we are adding a setter. I see there are one or two setters but we also have HFileContextBuilder. Shouldn't we be going via the Builder making HFileContexts? And why is this method not added on the Builder? (Can it be added non-public to encourage users to go via the Builder)?

Reply:
fixed, and reverted back to builder pattern

Reply:
You haven't uploaded new PR so will leave this as unresovled for now.

Reply:
oops, I missed this change. and now it should have updated/removed.